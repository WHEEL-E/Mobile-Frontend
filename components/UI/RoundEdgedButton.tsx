import React from "react";
import { StyleSheet, View, TouchableOpacity, Text } from "react-native";

interface roundEdgedButtonProps {
  title: string;
  buttonStyle: object;
  titleStyle: object;
  onPress: () => void;
}

const RoundEdgedButton = (props: roundEdgedButtonProps) => {
  const { title, buttonStyle, titleStyle, onPress } = props;
  return (
    <View>
      <TouchableOpacity
        style={{ ...buttonStyle, ...styles.button }}
        onPress={onPress}
      >
        <Text style={titleStyle}>{title}</Text>
      </TouchableOpacity>
    </View>
  );
};

export default RoundEdgedButton;

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    height: 55,
  },
});
