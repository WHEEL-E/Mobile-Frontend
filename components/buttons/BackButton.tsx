import React from "react";
import { StyleSheet, TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { DEVICE_WIDTH } from "../../utilities/constants/dimentions";
import { InputBackButtonProps } from "../../utilities/buttonsUtils";

export const BackButton = (props: InputBackButtonProps) => {
  const { onPress } = props;

  return (
    <TouchableOpacity style={styles.button}>
      <Ionicons name="arrow-back" color="#3F414E" size={35} onPress={onPress} />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    width: DEVICE_WIDTH * 0.15,
    height: DEVICE_WIDTH * 0.15,
    borderRadius: DEVICE_WIDTH * 0.075,
    alignItems: "center",
    justifyContent: "center",
    borderColor: "#EBEAEC",
    borderWidth: 0.7,
  },
});
