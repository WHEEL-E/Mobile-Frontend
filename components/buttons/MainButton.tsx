import React from "react";
import { StyleSheet, View, TouchableOpacity, Text, Image } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { MainButtonProps } from "../../utilities/buttonsUtils";

export const MainButton = (props: MainButtonProps) => {
  const { title, buttonStyle, titleStyle, onPress, icon, image } = props;

  return (
    <TouchableOpacity
      style={{ ...styles.button, ...buttonStyle }}
      onPress={onPress}
    >
      <View style={styles.imageContainer}>
        {icon && (
          <Ionicons name={icon.name} size={icon.size} color={icon.color} />
        )}
        {image && <Image source={image.url} style={{ ...styles.image }} />}
      </View>
      <Text style={titleStyle}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
  },
  imageContainer: {
    width: "60%",
    height: "60%",
    justifyContent: "center",
    alignItems: "center",
  },
  image: {
    width: "100%",
    height: "100%",
    resizeMode: "center",
  },
});
