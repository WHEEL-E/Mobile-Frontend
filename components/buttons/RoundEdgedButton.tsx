import React from "react";
import { StyleSheet, TouchableOpacity, Text } from "react-native";
import { roundEdgedButtonProps } from "../../utilities/types/buttonTypes";
import { DEVICE_WIDTH } from "../../utilities/constants/dimentions";

const RoundEdgedButton = (props: roundEdgedButtonProps) => {
  const { title, backgroundColor, onPress } = props;
  return (
    <TouchableOpacity
      onPress={onPress}
      style={{ ...styles.button, backgroundColor }}
    >
      <Text style={styles.titleStyle}>{title}</Text>
    </TouchableOpacity>
  );
};

export default RoundEdgedButton;

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    height: 55,
    width: "80%",
    borderWidth: 2,
    borderColor: "white",
    borderRadius: 30,
    backgroundColor: "green",
    alignSelf: "center",
  },
  titleStyle: {
    fontFamily: "Cairo-Bold",
    color: "white",
    fontSize: DEVICE_WIDTH * 0.05,
  },
});
