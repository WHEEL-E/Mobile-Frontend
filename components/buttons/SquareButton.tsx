import React from "react";
import { StyleSheet, TouchableOpacity, Text } from "react-native";
import { SquareButtonProps } from "../../utilities/types/buttonTypes";

export const SquareButton = (props: SquareButtonProps) => {
  const { title, buttonStyle, titleStyle, onPress } = props;

  return (
    <TouchableOpacity
      style={{ ...styles.button, ...buttonStyle }}
      onPress={onPress}
    >
      <Text style={titleStyle}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    borderWidth: 2,
    borderColor: "white",
    borderRadius: 10,
  },
});
