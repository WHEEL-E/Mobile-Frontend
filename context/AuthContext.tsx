import * as React from "react";
import { useDispatch } from "react-redux";
import { signIn, signOut } from "../store/actions/signIn";
import {
  AuthContextPlaceHolder,
  AuthProviderProps,
  SignInData,
} from "../utilities/types/signInTypes";

const AuthContext = React.createContext(AuthContextPlaceHolder);

export const AuthProvider = (props: AuthProviderProps) => {
  const { children } = props;
  const dispatch = useDispatch();

  const authFunctions = {
    signIn: async (data: SignInData) => {
      dispatch(signIn(data));
    },
    signOut: () => dispatch(signOut()),
    signUp: async (data: SignInData) => {
      dispatch(signIn(data));
    },
  };

  const authContext = React.useMemo(() => authFunctions, []);

  return (
    <AuthContext.Provider value={authContext}>{children}</AuthContext.Provider>
  );
};

export function useAuth() {
  const context = React.useContext(AuthContext);
  if (context === undefined) {
    throw new Error("useSignIn must be used within a signInProvider");
  }
  return context;
}
