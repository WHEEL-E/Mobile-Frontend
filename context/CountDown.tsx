import { useEffect, useMemo, useState } from "react";
import { getReturnValues } from "../utilities/types/sentInvitationsTypes";

const useCountdown = (targetDate: any) => {
  const sevenDays = 7 * 24 * 60 * 60 * 1000;
  const countDownDate = useMemo(
    () => sevenDays + new Date(targetDate).getTime() - Date.now(),
    []
  );

  const [countDown, setCountDown] = useState(countDownDate);

  useEffect(() => {
    const interval = setInterval(() => {
      setCountDown(
        (t) => sevenDays + new Date(targetDate).getTime() - Date.now()
      );
    }, 1000);
    if (countDown <= 0) {
      clearInterval(interval);
    }
    return () => clearInterval(interval);
  }, [countDownDate]);

  return getReturnValues(countDown);
};

export { useCountdown };
