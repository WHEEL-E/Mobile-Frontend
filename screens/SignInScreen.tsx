import React from "react";
import {
  Image,
  ImageBackground,
  Keyboard,
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
  View,
} from "react-native";
import { useTranslation } from "react-i18next";
import colors from "../utilities/constants/colors";
import { SignInProps } from "../utilities/types/navigationTypes/getStartedNavigationTypes";
import { DEVICE_WIDTH } from "../utilities/constants/dimentions";
import {
  HeadingText,
  NormalText,
  NoteText,
} from "../utilities/types/fontTypes";
import {
  PADDING_HORIZONTAL,
  PADDING_VERTICAL,
  SMALL_MARGIN_VERTICAL,
} from "../utilities/constants/spacing";
import SignInForm from "../components/signInComponents/SignInForm";

const SignInScreen = (props: SignInProps) => {
  const { t } = useTranslation();
  const { navigation } = props;

  const [isKeyBoardShown, setIsKeyBoardShown] = React.useState(false);

  React.useEffect(() => {
    const showSubscription = Keyboard.addListener("keyboardDidShow", () => {
      setIsKeyBoardShown(true);
    });
    const hideSubscription = Keyboard.addListener("keyboardDidHide", () => {
      setIsKeyBoardShown(false);
    });

    return () => {
      showSubscription.remove();
      hideSubscription.remove();
    };
  }, []);

  let padding = {
    paddingTop: "30%",
    paddingBottom: "10%",
  };

  if (isKeyBoardShown) {
    padding = {
      paddingTop: "15%",
      paddingBottom: "5%",
    };
  } else {
    padding = {
      paddingTop: "30%",
      paddingBottom: "10%",
    };
  }

  return (
    <View style={styles.container}>
      <TouchableWithoutFeedback
        style={styles.container}
        onPress={Keyboard.dismiss}
      >
        <ImageBackground
          source={require("../assets/images/vector-back.png")}
          style={{ ...styles.backgroundImage, ...padding }}
          resizeMode="contain"
          imageStyle={styles.imageStyle}
          resizeMethod="auto"
        >
          <View style={{ width: "100%", alignItems: "center" }}>
            <Image
              style={styles.logo}
              source={require("../assets/images/logo-b-app.png")}
            />
            <Text style={HeadingText}>{t("signInScreen.WelcomeBack")}</Text>
          </View>
          <SignInForm />
          <TouchableOpacity>
            <Text style={styles.forgotPasswordText}>
              {t("signInScreen.forgotPassword")}
            </Text>
          </TouchableOpacity>
          {!isKeyBoardShown && (
            <View style={styles.signUp}>
              <Text style={styles.notMemberText}>
                {t("signInScreen.notMember")}
              </Text>
              <TouchableOpacity onPress={() => navigation.navigate("SignUp")}>
                <Text style={styles.signUpText}>
                  {t("signInScreen.signUp")}
                </Text>
              </TouchableOpacity>
            </View>
          )}
        </ImageBackground>
      </TouchableWithoutFeedback>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "white",
  },
  backgroundImage: {
    alignItems: "center",
    flex: 1,
    alignSelf: "center",
    width: "100%",
  },
  imageStyle: {
    borderColor: "red",
    position: "absolute",
    top: 0,
    width: DEVICE_WIDTH,
    height: DEVICE_WIDTH * 0.7,
  },
  backButton: {
    position: "absolute",
    top: "7%",
    left: "8%",
  },
  logo: {
    height: 40,
    resizeMode: "center",
    width: "70%",
    alignSelf: "center",
  },
  inputField: {
    width: "80%",
    height: 70,
    maxHeight: "30%",
    marginBottom: SMALL_MARGIN_VERTICAL * 2,
  },
  inputs: {
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
    flex: 1,
  },
  buttons: {
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
  },
  forgotPasswordText: {
    ...NoteText,
    marginVertical: SMALL_MARGIN_VERTICAL,
    paddingHorizontal: PADDING_HORIZONTAL,
  },
  signUp: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    padding: PADDING_VERTICAL,
  },
  notMemberText: {
    ...NormalText,
    paddingHorizontal: PADDING_HORIZONTAL,
  },
  signUpText: {
    ...NormalText,
    paddingHorizontal: PADDING_HORIZONTAL,
    color: colors.darkGreen,
  },
  validationText: {
    color: "red",
    ...NoteText,
  },
});

export default SignInScreen;
