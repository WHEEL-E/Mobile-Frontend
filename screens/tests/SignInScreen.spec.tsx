import { render } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import SignInScreen from "../SignInScreen";
import * as roundEdgedButton from "../../components/buttons/RoundEdgedButton";
import { RoundEdgedButton } from "../../components/buttons/RoundEdgedButton";
import { AuthProvider } from "../../context/AuthContext";
import { Provider } from "react-redux";
import { applyMiddleware, createStore } from "redux";
import reducer from "../../store/reducers/rootReducer";
import ReduxThunk from "redux-thunk";
import React, { useState } from "react";

describe("SignInScreen", () => {
  const store = createStore(reducer, applyMiddleware(ReduxThunk));
  const dummyProp: any = () => {};
  const Screen = (
    <Provider store={store}>
      <AuthProvider>
        <SignInScreen navigation={dummyProp} route={dummyProp} />
      </AuthProvider>
    </Provider>
  );

  it("makes sure screen matches snapShot", () => {
    const signInScreen = renderer.create(Screen).toJSON();
    expect(signInScreen).toMatchSnapshot();
  });

  it("makes sure button is normally shown", () => {
    jest.spyOn(roundEdgedButton, "RoundEdgedButton");
    render(Screen);
    expect(RoundEdgedButton).toBeCalledTimes(1);
  });

  it("makes sure button disappears when keyboard is shown ", () => {
    const setStateMock = jest.fn();
    const useStateMock: any = (useState: any) => [true, setStateMock];
    jest.spyOn(React, "useState").mockImplementation(useStateMock);
    jest.spyOn(roundEdgedButton, "RoundEdgedButton");

    render(Screen);
    expect(RoundEdgedButton).toBeCalledTimes(0);
  });
});
