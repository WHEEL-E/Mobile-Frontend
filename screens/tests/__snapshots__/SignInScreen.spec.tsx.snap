// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInScreen makes sure screen matches snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingVertical": "30%",
        "resizeMode": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "height": 40,
          "resizeMode": "center",
          "width": "70%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "Cairo-Bold",
          "fontSize": 27,
        }
      }
    >
      signInScreen.WelcomeBack
    </Text>
  </View>
  <View
    style={
      Object {
        "left": "8%",
        "position": "absolute",
        "top": "7%",
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#EBEAEC",
          "borderRadius": 56.25,
          "borderWidth": 0.7,
          "height": 112.5,
          "justifyContent": "center",
          "opacity": 1,
          "width": 112.5,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="email"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="signInScreen.emailAddress"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#F2F3F7",
          "borderRadius": 15,
          "height": 70,
          "marginBottom": 15,
          "padding": 20,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCompleteType="password"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="signInScreen.password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#F2F3F7",
          "borderRadius": 15,
          "height": 70,
          "marginBottom": 15,
          "padding": 20,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#11698E",
            "borderColor": "white",
            "borderRadius": 30,
            "borderWidth": 2,
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-Bold",
              "fontSize": 37.5,
            }
          }
        >
          signInScreen.logIn
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Cairo-Regular",
              "fontSize": 26.68,
              "paddingHorizontal": 5,
            }
          }
        >
          signInScreen.forgotPassword
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Cairo-Regular",
          "fontSize": 33.35,
          "paddingHorizontal": 5,
        }
      }
    >
      signInScreen.notMember
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#11698E",
            "fontFamily": "Cairo-Regular",
            "fontSize": 33.35,
            "paddingHorizontal": 5,
          }
        }
      >
        signInScreen.signUp
      </Text>
    </View>
  </View>
</View>
`;
